connectors/sources/notion.py

```diff
@@ -158,18 +163,138 @@ async def fetch_comments(self, block_id):
         ):
             yield block_comment
 
-    async def query_database(self, database_id):
+    async def query_database(self, database_id, body=None):
+        if body is None:
+            body = {}
         async for result in async_iterate_paginated_api(
-            self._get_client.databases.query, database_id=database_id
+            self._get_client.databases.query, database_id=database_id, **body
         ):
             yield result
 
 
+class NotionAdvancedRulesValidator(AdvancedRulesValidator):
+    DATABASE_QUERY_DEFINITION = {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "database_id": {"type": "string", "minProperties": 1},
+                "filter": {
+                    "type": "object",
+                    "properties": {"property": {"type": "string", "minProperties": 1}},
+                    "additionalProperties": {
+                        "type": ["object", "array"],
+                        "minProperties": 1,
+                    },
+                },
+            },
+            "required": ["database_id"],
+        },
+    }
+
+    SEARCH_QUERY_DEFINITION = {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "query": {"type": "string"},
+                "filter": {
+                    "type": "object",
+                    "properties": {
+                        "value": {"type": "string", "enum": ["page", "database"]},
+                    },
+                    "required": ["value"],
+                },
+            },
+            "required": ["query"],
+        },
+    }
+
+    RULES_OBJECT_SCHEMA_DEFINITION = {
+        "type": "object",
+        "properties": {
+            "database_query_filters": DATABASE_QUERY_DEFINITION,
+            "searches": SEARCH_QUERY_DEFINITION,
+        },
+        "minProperties": 1,
+        "additionalProperties": False,
+    }
+
+    SCHEMA = fastjsonschema.compile(definition=RULES_OBJECT_SCHEMA_DEFINITION)
+
+    def __init__(self, source):
+        self.source = source
+
+    async def validate(self, advanced_rules):
+        if len(advanced_rules) == 0:
```

Comment:
Please add some logging to this function, so that we'll be able to troubleshoot what went wrong if something goes wrong.